src/traces/carpet/calc.js

```diff
@@ -52,8 +52,8 @@ module.exports = function calc(gd, trace) {
     trace.setScale();
 
     // Convert cartesian-space x/y coordinates to screen space pixel coordinates:
-    t.xp = trace.xp = map2dArray(trace.xp, x, xa.c2p);
-    t.yp = trace.yp = map2dArray(trace.yp, y, ya.c2p);
+    t.xp = trace._xp = map2dArray(trace._xp, x, xa.c2p);
+    t.yp = trace._yp = map2dArray(trace._yp, y, ya.c2p);
```

Comment:
Are these xp and yp values used anywhere? Here they appear to be reset during the plot step.

Reply:
ha good call. Nothing fails if you 🔪 these lines, which is good because the `calc` step had *better* not generate pixel positions, the axis scaling isn't known yet at that point!
Removed in https://github.com/plotly/plotly.js/pull/2525/commits/0221510d0e1443aec95f7830659cfdffcd6e6772

Reply:
wow, then `map_2d_array` became totally unused -> https://github.com/plotly/plotly.js/pull/2525/commits/b0af416a3561da25af1fa726da99be30de1b6c68